File: 5-Problem-A-Day/228. Summary Ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> ans = new LinkedList<>();
        if (nums.length == 0)
            return ans;
        int start = nums[0], last = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] - last != 1) {
                if (last == start)
                    ans.add(Integer.toString(start));
                else
                    ans.add(Integer.toString(start) + "->" + Integer.toString(last));
                start = last = nums[i];
            } else
                last = nums[i];
        }
        if (last == start)
            ans.add(Integer.toString(start));
        else
            ans.add(Integer.toString(start) + "->" + Integer.toString(last));
        return ans;
    }
}

File: 2024/228. Summary Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []
        start = end = nums[0]
        ans = list()
        for i in range(1, len(nums)):
            if nums[i] - nums[i - 1] == 1:
                end = nums[i]
            else:
                if start == end:
                    ans.append(str(start))
                else:
                    ans.append(str(start)+"->"+str(end))
                start = end = nums[i]
        if start == end:
            ans.append(str(start))
        else:
            ans.append(str(start)+"->"+str(end))
        return ans


class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []
        ans = list()
        i = 0
        while i < len(nums):
            start = nums[i]
            while i + 1 < len(nums) and nums[i] + 1 == nums[i + 1]:
                i += 1
            if start != nums[i]:
                ans.append(str(start) + "->" + str(nums[i]))
            else:
                ans.append(str(start))
            i += 1
        return ans
